### ISSUE TITLE
RingbufferAsyncAddWithBackoffStressTest.whenShortTTLAndBigBuffer

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#7193]
```
java.lang.AssertionError: No error should have been thrown, but ConsumeThread-1 completed error expected null, but was:<com.hazelcast.ringbuffer.StaleSequenceException: sequence:5814 is too small. The current headSequence is:5818>
    at org.junit.Assert.fail(Assert.java:88)
    at org.junit.Assert.failNotNull(Assert.java:755)
    at org.junit.Assert.assertNull(Assert.java:737)
    at com.hazelcast.test.TestThread.assertSucceedsEventually(TestThread.java:60)
    at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.test(RingbufferAsyncAddWithBackoffStressTest.java:81)
    at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.whenShortTTLAndBigBuffer(RingbufferAsyncAddWithBackoffStressTest.java:57)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-nightly/com.hazelcast$hazelcast/709/testReport/junit/com.hazelcast.ringbuffer.impl/RingbufferAsyncAddWithBackoffStressTest/whenShortTTLAndBigBuffer/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: No error should have been thrown, but ConsumeThread-1 completed error expected null, but was:<com.hazelcast.ringbuffer.StaleSequenceException: sequence:4805 is too small. The current headSequence is:4807 tailSequence is:4806>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at com.hazelcast.test.TestThread.assertSucceedsEventually(TestThread.java:62)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.test(RingbufferAsyncAddWithBackoffStressTest.java:81)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.whenShortTTLAndBigBuffer(RingbufferAsyncAddWithBackoffStressTest.java:57)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/1010/testReport/junit/com.hazelcast.ringbuffer.impl/RingbufferAsyncAddWithBackoffStressTest/whenShortTTLAndBigBuffer/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: No error should have been thrown, but ConsumeThread-2 completed error expected null, but was:<com.hazelcast.ringbuffer.StaleSequenceException: sequence:441 is too small. The current headSequence is:442 tailSequence is:441>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at com.hazelcast.test.TestThread.assertSucceedsEventually(TestThread.java:62)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.test(RingbufferAsyncAddWithBackoffStressTest.java:82)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.whenShortTTLAndBigBuffer(RingbufferAsyncAddWithBackoffStressTest.java:57)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/1013/testReport/junit/com.hazelcast.ringbuffer.impl/RingbufferAsyncAddWithBackoffStressTest/whenShortTTLAndBigBuffer/

**Comment by vbekiaris** [comment]
This does not imply a possible correctness issue, rather it is a stress test and consumers may sporadically fall behind the producer due to unpredictable load in CI environment.

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: No error should have been thrown, but ConsumeThread-1 completed error expected null, but was:<com.hazelcast.ringbuffer.StaleSequenceException: sequence:736 is too small. The current headSequence is:737 tailSequence is:736>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at com.hazelcast.test.TestThread.assertSucceedsEventually(TestThread.java:62)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.test(RingbufferAsyncAddWithBackoffStressTest.java:81)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.whenShortTTLAndBigBuffer(RingbufferAsyncAddWithBackoffStressTest.java:57)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/1082/testReport/junit/com.hazelcast.ringbuffer.impl/RingbufferAsyncAddWithBackoffStressTest/whenShortTTLAndBigBuffer/

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/1084/testReport/junit/com.hazelcast.ringbuffer.impl/RingbufferAsyncAddWithBackoffStressTest/whenShortTTLAndBigBuffer/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: No error should have been thrown, but ConsumeThread-2 completed error expected null, but was:<com.hazelcast.ringbuffer.StaleSequenceException: sequence:384796 is too small and data store is disabled. The current headSequence is:384797 tailSequence is:384796>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at com.hazelcast.test.TestThread.assertSucceedsEventually(TestThread.java:62)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.test(RingbufferAsyncAddWithBackoffStressTest.java:82)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.whenShortTTLAndBigBuffer(RingbufferAsyncAddWithBackoffStressTest.java:57)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/1105/testReport/junit/com.hazelcast.ringbuffer.impl/RingbufferAsyncAddWithBackoffStressTest/whenShortTTLAndBigBuffer/

**Comment by jerrinot** [comment]
fixed by #7193

**Comment by hasancelik** [comment]
*master*
```
org.junit.runners.model.TestTimedOutException: test timed out after 300000 milliseconds
	at java.lang.Thread.sleep(Native Method)
	at java.lang.Thread.sleep(Thread.java:340)
	at java.util.concurrent.TimeUnit.sleep(TimeUnit.java:360)
	at com.hazelcast.test.HazelcastTestSupport.sleepSeconds(HazelcastTestSupport.java:269)
	at com.hazelcast.test.HazelcastTestSupport.sleepAndStop(HazelcastTestSupport.java:291)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.test(RingbufferAsyncAddWithBackoffStressTest.java:78)
	at com.hazelcast.ringbuffer.impl.RingbufferAsyncAddWithBackoffStressTest.whenShortTTLAndBigBuffer(RingbufferAsyncAddWithBackoffStressTest.java:58)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/1254/testReport/junit/com.hazelcast.ringbuffer.impl/RingbufferAsyncAddWithBackoffStressTest/whenShortTTLAndBigBuffer/

**Comment by ahmetmircik** [comment]
This seems a fix for this issue: 74f39ed
